src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/hmrcAddToAList.scala.html

```diff
@@ -0,0 +1,104 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.govukfrontend.views.Aliases.Hint
+@import uk.gov.hmrc.govukfrontend.views.html.components.{GovukButton, GovukRadios}
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.button.Button
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.content.Text
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.fieldset.{Fieldset, Legend}
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.radios.{RadioItem, Radios}
+@import uk.gov.hmrc.hmrcfrontend.views.helpers.MessageHelpers.{messagesForLanguage, pluralMessageKey}
+@import uk.gov.hmrc.hmrcfrontend.views.viewmodels.addtoalist.AddToAList
+
+@this(govukRadios: GovukRadios, govukButton: GovukButton)
+
+@(params: AddToAList)(implicit messages: Messages, messagesApi: MessagesApi)
+@import params._
+
+@defining(messagesForLanguage(language)) { messages =>
+
+    <h1 class="govuk-heading-xl">
+        @messages(
+            pluralMessageKey("addToAList.heading", itemList.length),
+            itemList.length,
+            itemType.map { itemType =>
+                itemList.length match {
+                    case 1 => itemType.singular
+                    case _ => itemType.plural
+                }
+            }.getOrElse {
+                messages(pluralMessageKey("addToAList.defaultItemType", itemList.length))
+            }
+        )
+    </h1>
+
+    <div class="govuk-form-group">
+        <dl class="hmrc-add-to-a-list hmrc-add-to-a-list--@itemSize.suffix">
+        @for(item <- itemList) {
+            <div class="hmrc-add-to-a-list__contents">
+                <dt class="hmrc-add-to-a-list__identifier hmrc-add-to-a-list__identifier--light">
+                @item.name
+                </dt>
+                <dd class="hmrc-add-to-a-list__change">
+                    <a class="govuk-link" href="@item.changeUrl">
+                        <span aria-hidden="true">@messages("addToAList.change.displayed")</span>
+                        <span class="govuk-visually-hidden">@messages("addToAList.change.announced", item.name)</span>
+                    </a>
+                </dd>
+                <dd class="hmrc-add-to-a-list__remove">
+                    <a class="govuk-link" href="@item.removeUrl">
+                        <span aria-hidden="true">@messages("addToAList.remove.displayed")</span>
+                        <span class="govuk-visually-hidden">@messages("addToAList.remove.announced", item.name)</span>
+                    </a>
+                </dd>
+            </div>
+        }
+        </dl>
+    </div>
+    <form method="post" action="@formAction">
```

Comment:
- [ ] going to raise seperately, or at least ask, should this be a post? Maybe the pattern is POST to create empty thing -> then edit it; I want to check because I'm more used to GET form to create new entity -> POST to create

Reply:
It's a state-changing request, so I would expect the method to be a POST and to protect against CSRF. I'm probably missing some context in your question though :-)

Reply:
my thought on this was is submitting this form actually going to create another thing? Or will it just take you to the form that you would then submit to add another (so the state change would be submitting that next form), feels like here we've not got an inline form for creating an "item" so much as we have a question about if you'd like to add another. I should probably see if I can find an example of it in use and talk to design system team - probably that would clear up intended use here

Reply:
I agree, I hadn't noticed before but at this stage you're essentially specifying which page you want to see next so a GET would be perfectly suitable.

I think there's an unspoken standard of these kind of actions being POST requests (e.g. https://www.tax.service.gov.uk/pay).  I'd suggest doing this one "properly" and use it as an example - safe and idempotent actions are GET requests, unsafe or non-idempotent actions are POST request.